### PR TITLE
vrtdataset: vrt connection option 'a_ullr'

### PR DESCRIPTION
**Pull request by mdsumner**
Adds 'a_ullr' to the allowed options for a "vrt:// connection. The effect is to override the georeferenced bounds in the same way as the option to gdal_translate.

## Examples

```
gdalinfo  "vrt://autotest/gdrivers/data/netcdf/srid.nc?a_ullr=0,90,360,-90"

## chain with other options
gdalinfo  "vrt://autotest/gdrivers/data/netcdf/srid.nc?a_ullr=0,1,1,0&bands=1,1"
```

## related  pull requests

#6893 

## Tasklist

 - [x] Add test case(s)
 - [x] Add documentation
 - [x] Review
 - [x] Adjust for comments
 - [x] review: use commas not spaces
 - [x] review: clarify order of options ulx,uly,lrx,lry (xmin, ymax, xmax, ymin)  
 - [x] review: avoid string/float churn of tokens 
 - [x] fix python assert  (!) 
 - [ ] All CI builds and checks have passed

### CODE REVIEW

**Comment by rouault** [comment]
> gdalinfo  "vrt://autotest/gdrivers/data/netcdf/srid.nc?a_ullr=0 90 360 -90"

looks weird with space separator. Comma would looks better, and consistent with the bands option.

File: frmts/vrt/vrtdataset.cpp [hunk 1]
```diff
@@ -1013,6 +1013,32 @@ GDALDataset *VRTDataset::OpenVRTProtocol( const char* pszSpec )
...
+              xmin = atof(aosUllr[0]);
+              xmax = atof(aosUllr[2]);
+              ymin = atof(aosUllr[3]);
+              ymax = atof(aosUllr[1]);
+              
+              argv.AddString("-a_ullr"); 
+              argv.AddString(CPLSPrintf("%f", xmin));
+              argv.AddString(CPLSPrintf("%f", ymax));
+              argv.AddString(CPLSPrintf("%f", xmax));
+              argv.AddString(CPLSPrintf("%f", ymin));
```
Comment (rouault) [inline frmts/vrt/vrtdataset.cpp:1040]: you don't actually need to do string -> floating-point -> string:
```suggestion
              argv.AddString("-a_ullr"); 
              argv.AddString(aosUllr[0]);
              argv.AddString(aosUllr[1]);
              argv.AddString(aosUllr[2]);
              argv.AddString(aosUllr[3]);
```

File: doc/source/drivers/raster/vrt.rst [hunk 2]
```diff
@@ -1622,24 +1622,35 @@ For example:
...
-in the same way as (:ref:`gdal_translate`), it may be missing, or incorrect. The value provided
-for 'a_srs' may be be a string or a file containing a srs definition.
+The effect of the ``a_srs`` option (added in GDAL 3.7) is to assign (override) the coordinate 
+reference system of the source in the same way as (:ref:`gdal_translate`), it may be missing, 
+or incorrect. The value provided for ``a_srs`` may be a string or a file containing a srs 
+definition.
+
+The effect of the ``a_ullr`` option (added in GDAL 3.7) is to assign (override) the georeferenced
+bounds of the source in the same way as (:ref:`gdal_translate`). The value consists of four numeric
+values separated by spaces.
```
[the remark this answers is not in the document]
Comment (mdsumner) [reply doc/source/drivers/raster/vrt.rst]: done

**Comment by mdsumner** [comment]
thanks @rouault I've addressed all now, the CI failing windows-conda but I can't tell if that's my fault

**Comment by rouault** [comment]
> the CI failing windows-conda but I can't tell if that's my fault

just a flake.

Merging

**Comment by mdsumner** [comment]
👌

**Pull request merged**